Makefile

```diff
@@ -15,20 +19,16 @@ ifneq ($(NO_BGW),)
 	MODULES=
 endif
 
+.PHONY: all
 all: sql/$(EXTENSION)--$(EXTVERSION).sql
 
 SCRIPTS = bin/common/*.py
 
-sql/$(EXTENSION)--$(EXTVERSION).sql: $(sort $(wildcard sql/types/*.sql)) $(sort $(wildcard sql/tables/*.sql)) $(sort $(wildcard sql/functions/*.sql)) $(sort $(wildcard sql/procedures/*.sql))
-	cat $^ > $@
+sql/$(EXTENSION)--$(EXTVERSION).sql: | $(wildcard sql/types/*.sql sql/tables/*.sql sql/functions/*.sql sql/procedures/*.sql)
+	cat $| > $@
```

Comment:
So I specifically had the sort operations on each of the individual types of files because I wanted a predictable way to have certain operations be first for each type of object.  Since extension updates/installs are all done as one giant transaction, this was the only way I could have to be able to guarantee a specific point that certain operations could run. I could put those operations in the alphabetically first file of that type and it would guarantee to be first.

Reply:
This was a necessity early on in some development. Don't think I've had to worry about this for a while, but I would like to keep the functionality there if needed

Reply:
Testing it out, it looks like the `$|` does the same thing. But just want to make sure the end result always the same as what the previous options did.

Reply:
The docs says[1] the returned list is sorted (the link is mentioned in the commit[2]):

> As with wildcard expansion in rules, the results of the wildcard function are sorted. But again, each individual expression is sorted separately, so ‘$(wildcard *.c *.h)’ will expand to all files matching ‘.c’, sorted, followed by all files matching ‘.h’, sorted. 

---
[1] https://www.gnu.org/software/make/manual/html_node/Wildcard-Function.html
[2] https://github.com/pgpartman/pg_partman/commit/525d92a51618f32b31b62a99fa73b122eb75ebb8

Reply:
> are never checked

Ah! I missed that sentence. OK, we can leave "$^", but still use a single "$(wildcard …)" call… 😇

Reply:
Please go back and re-read my initial reasoning for the sorting being there. I would like to keep the functionality to allow a predictable method of running code at a specific point during an update or install.

Reply:
Hmm. If I understand docs correctly, there is nothing changed in sorting order.

The only reason why I tried to use "$|" was an attempt to [possible] guard against multijob (`make -j5`). But if `make` will sort prerequisites by the time when they are ready without a pipe symbol it will do it for prerequisites with your multiple `$(sort …)` entries because they are just gives a list of required targets which will be the same as when using a single `$(wildcard …)` and multiple rules inside it.

Please could you show how my change leads to a different result.

Reply:
P.S.: I'd like to use the single "$(wildcard …)" because its behavior is the same as you wrote in the initial reasoning.

I posted an excerpt from docs about expected behavior in this discussion above:
https://github.com/pgpartman/pg_partman/pull/546#discussion_r1253295553

Reply:
Hmm ok I missed that that was part of the wildcard doc itself and not the `$|` entry.

Are you sure that's consistent for all make or just gnu make? I could've sworn the reason I did that was because I was just getting arbitrary file name lists back with no predictable ordering.

Reply:
Ok. I doubt PG is making use of the actual order that things are put together in any significant way. Haven't had to worry about the order for a long time, so I guess if it ever comes up again in the future I'll explore it again